build.sbt

```diff
@@ -163,6 +163,8 @@ lazy val http = project("http")
       "scaladsl/server/directives/FormFieldDirectives.scala", "scaladsl/server/directives/RespondWithDirectives.scala"))
   .enablePlugins(BootstrapGenjavadoc, BoilerplatePlugin)
   .enablePlugins(ReproducibleBuildsPlugin)
+  .settings(
+    Compile / headerSources ++= (Compile / boilerplateSource).value.**("*.scala.template").get)
```

Comment:
Most of these `headerSources` configs appear in `project/CopyrightHeader*.scala`. Does this need to be here because the `boilerplateSource` setting is only available here?

Reply:
`boilerplateSource` is avalable only in `http` project which uses `BoilerplatePlugin`.

But it's possible to use it before enable plugin, so move this settings in other `.settings` method.

https://github.com/apache/incubator-pekko-http/pull/258/commits/d17bd7f9111ca30da7f4116e741f30d1fb2162d1

If we don't need to use `boilerplateSource`, we can add settings into aggregated project. It means we can put settings with `*.scala.template` only, not `boilerplateSource`.

Reply:
I would agree with @pjfanning here, can we move the settings into `project/CopyrightHeaders.scala`. While it is true that right now the boilerplate may only exist in a single project, there is no harm in making it in global and if future `.scala.template` files are created in other projects/they are moved around it will catch them.

Remember the underlying context behind all of these header checking, its much better to be more aggressive and if there are any false positives to handle/deal with it rather than the other way around.

Reply:
I see. Now `project/CopyrightHeaders.scala` works fine.

https://github.com/apache/incubator-pekko-http/pull/258/commits/b39ef0d2706572e1429bc59954d13b818a408eed


```
✘╹◡╹✘☆  sbt 'show http/headerSources' | tail -n 10
[info] * /Users/krrrr38/dev/src/github.com/apache/incubator-pekko-http/http/src/main/scala/org/apache/pekko/http/scaladsl/server/directives/FileAndResourceDirectives.scala
[info] * /Users/krrrr38/dev/src/github.com/apache/incubator-pekko-http/http/src/main/scala/org/apache/pekko/http/scaladsl/marshalling/PredefinedToEntityMarshallers.scala
[info] * /Users/krrrr38/dev/src/github.com/apache/incubator-pekko-http/http/src/main/boilerplate/org/apache/pekko/http/scaladsl/server/directives/FormFieldDirectivesInstances.scala.template
[info] * /Users/krrrr38/dev/src/github.com/apache/incubator-pekko-http/http/src/main/boilerplate/org/apache/pekko/http/scaladsl/server/directives/ParameterDirectivesInstances.scala.template
[info] * /Users/krrrr38/dev/src/github.com/apache/incubator-pekko-http/http/src/main/boilerplate/org/apache/pekko/http/scaladsl/server/util/ApplyConverterInstances.scala.template
[info] * /Users/krrrr38/dev/src/github.com/apache/incubator-pekko-http/http/src/main/boilerplate/org/apache/pekko/http/scaladsl/server/util/ConstructFromTupleInstances.scala.template
[info] * /Users/krrrr38/dev/src/github.com/apache/incubator-pekko-http/http/src/main/boilerplate/org/apache/pekko/http/scaladsl/server/util/TupleFoldInstances.scala.template
[info] * /Users/krrrr38/dev/src/github.com/apache/incubator-pekko-http/http/src/main/boilerplate/org/apache/pekko/http/scaladsl/server/util/TupleAppendOneInstances.scala.template
[info] * /Users/krrrr38/dev/src/github.com/apache/incubator-pekko-http/http/src/main/boilerplate/org/apache/pekko/http/javadsl/server/JavaPathMatchers.scala.template
[success] Total time: 0 s, completed 2023 Jul 23 17:04:41
```

Reply:
Thanks! Running CI